Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -25,6 +25,19 @@ public IonObjectSerializer(IonSerializer ionSerializer, IonSerializationOptions
 
         public override object Deserialize(IIonReader reader)
         {
+            var ionConstructors = targetType.GetConstructors(BINDINGS).Where(IsIonConstructor);
```

Comment:
You might want to add Take(2) at the end, for possible performance gain (in case there are more than two constructors).

Reply:
Done.